lib/spack/spack/spec.py

```diff
@@ -2244,6 +2244,21 @@ def concretize(self, tests=False):
         # Mark everything in the spec as concrete, as well.
         self._mark_concrete()
 
+        # If any spec in the DAG is deprecated, throw an error
```

Comment:
Why is this an error? In the openssl case some packages will be attached to the old openssl and as the installation DB is updated, new ones will likely need it too. Should this just be a warning message?

Reply:
New specs should be force to build against the new spec, not the old one. If a spec is deprecated we've already said no one should use it, so this error is to enforce that.